app/src/test/java/org/simple/clinic/sync/SyncDataOnApprovalTest.kt

```diff
@@ -28,18 +40,40 @@ class SyncDataOnApprovalTest {
     RxJavaPlugins.setIoSchedulerHandler { Schedulers.trampoline() }
   }
 
-  @Parameters(value = ["true", "false"])
+  @Parameters(method = "params for syncing on user status changed")
   @Test
-  fun `when user is allowed to sync data only then data sync should be initiated`(canSync: Boolean) {
-    whenever(userSession.canSyncData()).thenReturn(Observable.just(canSync))
+  fun `when the user status changes to ApprovedFromSyncing from anything, the sync should be triggered`(previousUserStatus: UserStatus) {
+    // given
+    val user = PatientMocker.loggedInUser(status = previousUserStatus)
+    val userSubject = PublishSubject.create<Optional<User>>()
+    whenever(userSession.loggedInUser()).thenReturn(userSubject)
     whenever(dataSync.sync(null)).thenReturn(Completable.complete())
 
+    // when
     syncDataOnApproval.sync()
 
-    if (canSync) {
-      verify(dataSync).sync(null)
-    } else {
-      verify(dataSync, never()).sync(null)
-    }
+    // then
+    userSubject.onNext(user.toOptional())
+    verifyZeroInteractions(dataSync)
+
+    userSubject.onNext(user.withStatus(ApprovedForSyncing, clock).toOptional())
+    verify(dataSync).sync(null)
```

Comment:
Curious, why does `sync` take in `null`?

Reply:
This is bad API design which we haven't gotten around to fixing yet. `null` here is a sentinel value which means "sync everything" instead of a specific sync group.

Reply:
I can take a stab at fixing this.

Reply:
Done.